npm-frontend/src/App.jsx

```diff
@@ -352,53 +202,71 @@ class App extends Component {
   // TODO (maybe): update the connection with firebase so we don't have to loop through all the events all the time 
   addGuest = () => {
     const eventList = [...this.state.eventList];
-    const thisEvent = this.state.thisEvent
+    const currEventCode = this.state.currEventCode    
 
     // Loop through each event in the event list
     eventList.forEach((e) => {
       // Check if the event code equals the current event
-      if(e.eventCode === thisEvent) {
+      if(e.eventCode === currEventCode) {
         // If guestList hasn't been created, create one
         if(e.guestList === undefined) {e.guestList = [];}
         // Add the name to the guestList
         e.guestList.push(this.state.name);
         // Update the eventList
         this.setState({ eventList });
 
-
+        console.log(this.state.name);
       }
     })
+    // Update the backend with the new user
+    fetch(`http://localhost:8080/api/add_user?eventCode=${this.state.currEventCode}&userId=${this.state.name}`);
   }
 
   // This is run when "Join Event" or "Create Event" are clicked on the homepage
   emptyLocalStorage = () => {
+    console.log('emptying storage')
     // Clear out any event code from local storage
     localStorage.clear();
+    this.setState({ host: false });
   }
 
-  getHashParams() {
-    let hashParams = {};
-    let e, r = /([^&;=]+)=?([^&;]*)/g,
-        q = window.location.hash.substring(1);
-    e = r.exec(q)
-    while (e) {
-       hashParams[e[1]] = decodeURIComponent(e[2]);
-       e = r.exec(q);
-    }
-    return hashParams;
+  updateToken = (token) => {
+    // Store the access token
+    localStorage.setItem('accessToken', JSON.stringify(token));
+    this.setState({ accessToken: token });
+  }
+
+  updateSongs = (data) => {
+    this.setState({ songData: data })
+    localStorage.setItem('songData', JSON.stringify(data));
+  }
+
+  endEvent = () => {
+    // Change the hostid if you want to run this with your own spotify account
+    fetch(`http://localhost:8080/api/end_event?eventCode=${this.state.currEventCode}&hostId=chipmilotis&accessToken=${this.state.accessToken}`);
```

Comment:
Is it possible to get the host's ID from the spotify authorization? If not I can throw something in the backend to do it